### PR TITLE
Support for running xfstests on loop device [v2]

### PR DESCRIPTION
**Pull request by harish-24**
Enables tests to be run on loop device
Runs all tests for given file system
(or)
Runs specific tests for given file system if range is provided.

Signed-off-by: Harish <harish@linux.vnet.ibm.com>

### CODE REVIEW

File: fs/xfstests.py.data/xfstests.yaml [hunk 1]
```diff
@@ -1,4 +1,19 @@
 setup:
-    runargs : !mux
-        skip_dangerous: True
-        test_range: '2,12,34-87,91,99-112'
+    skip_dangerous: True
+    file_system: 'ext4'
```
Comment (narasimhan-v) [inline fs/xfstests.py.data/xfstests.yaml]: @harish-24 did not notice this first. Can you please change this parameter name as `fs` to be uniform across all tests.

Comment (harish-24) [inline fs/xfstests.py.data/xfstests.yaml]: sure

File: fs/xfstests.py [hunk 2]
```diff
@@ -84,42 +100,87 @@ def setUp(self):
...
+        # In case if any test has been interrupted
+        process.system('umount %s %s' % (self.scratch_mnt, self.test_mnt),
+                       sudo=True, ignore_status=True)
+        if os.path.exists(self.scratch_mnt):
+            os.rmdir(self.scratch_mnt)
+        if os.path.exists(self.test_mnt):
+            os.rmdir(self.test_mnt)
+        if self.dev_type == 'loop':
+            for dev in self.loop_devices:
+                process.system('losetup -d %s' % dev, shell=True, sudo=True)
```
Comment (PraveenPenguin) [inline fs/xfstests.py]: please use `ignore_status=True` as parameter  in `process.system`

File: fs/xfstests.py [hunk 3]
```diff
@@ -84,42 +100,87 @@ def setUp(self):
...
+        self.part.unmount()
+
+    def _create_loop_device(self, base_disk):
+        self.part = partition.Partition(
+            base_disk, mountpoint=self.disk_mnt)
+        self.part.mount()
+        if process.system('which mkfs.%s' % self.fs_to_test):
```
Comment (PraveenPenguin) [inline fs/xfstests.py]: here also please

File: fs/xfstests.py [hunk 4]
```diff
@@ -84,42 +100,87 @@ def setUp(self):
...
+            self.skip('Unknown filesystem %s' % self.fs_to_test)
+        # Creating two loop devices
+        for i in range(2):
+            process.run('fallocate -o 0 -l 9GiB %s/file-%s.img' %
```
Comment (PraveenPenguin) [inline fs/xfstests.py]: can  9G Size also be handle via Yaml file ?

File: fs/xfstests.py.data/xfstests.yaml [hunk 5]
```diff
@@ -1,4 +1,19 @@
 setup:
-    runargs : !mux
-        skip_dangerous: True
-        test_range: '2,12,34-87,91,99-112'
+    skip_dangerous: True
+    fs: 'ext4'
+    scratch_mnt: '/mnt/scratch'
+    test_mnt: '/mnt/test'
+    disk_mnt: '/mnt/loop-device'
+
```
Comment (PraveenPenguin) [inline fs/xfstests.py.data/xfstests.yaml]: please remove this white space

File: fs/xfstests.py.data/xfstests.yaml [hunk 6]
```diff
@@ -1,4 +1,19 @@
...
+    # Uncomment and edit test_range for running specific tests
+    # test_range: '73,217-415'
+
```
Comment (PraveenPenguin) [inline fs/xfstests.py.data/xfstests.yaml]: here as well

File: fs/xfstests.py.data/README [hunk 7]
```diff
@@ -42,7 +44,4 @@ virtual partition depends on the following programs to be available:
 Make sure you have them or a real spare device to test things.
 """
 TODO:
-1- Device will be given as yaml input
-2- creating FS, mounting the disk/loop device
-3- Write local.cofig dynamic way 
-4- test will able to run only specfic file system type test case
+1- Replace local.cofig for real disks
```
Comment (PraveenPenguin) [inline fs/xfstests.py.data/README]: ok

**Comment by PraveenPenguin** [comment]
@harish-24 can you please attach job log as well

**Comment by PraveenPenguin** [comment]
@harish-24 can you please add your Authorship also as i saw you did significant change

**Comment by harish-24** [comment]
@PraveenPenguin pushed with the changes. Thanks !

**Comment by PraveenPenguin** [comment]
@narasimhan-v @nasastry thanks for review

**Pull request merged**